modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/fragment/renderer/DynamicFieldFragmentRenderer.java

```diff
@@ -0,0 +1,353 @@
+
+			if (Validator.isNull(namespace)) {
+				PortletDisplay portletDisplay =
+					themeDisplay.getPortletDisplay();
+
+				namespace = portletDisplay.getNamespace();
+
+				httpServletRequest.setAttribute(
+					"liferay-commerce:dynamic-field:namespace", namespace);
+			}
+
+			requestDispatcher.include(httpServletRequest, httpServletResponse);
+		}
+		catch (Exception exception) {
+			throw new RuntimeException(exception);
+		}
+	}
+
+	private String _getConfigurationValue(
+		FragmentEntryLink fragmentEntryLink, String name) {
+
+		return GetterUtil.getString(
+			_fragmentEntryConfigurationParser.getFieldValue(
+				fragmentEntryLink.getConfiguration(),
+				fragmentEntryLink.getEditableValues(),
+				LocaleUtil.getMostRelevantLocale(), name));
+	}
+
+	private String _getCPInstanceFieldValue(
+			HttpServletRequest httpServletRequest, CPDefinition cpDefinition,
+			String field)
+		throws PortalException {
+
+		CPInstanceHelper cpInstanceHelper =
+			ServletContextUtil.getCPInstanceHelper();
+
+		try {
+			CPInstance cpInstance = cpInstanceHelper.getDefaultCPInstance(
+				cpDefinition.getCPDefinitionId());
+
+			if (field.equals("availability.stockQuantity")) {
+				CommerceChannel commerceChannel =
+					_commerceChannelLocalService.
+						fetchCommerceChannelBySiteGroupId(
+							_portal.getScopeGroupId(httpServletRequest));
+
+				CommerceCatalog commerceCatalog =
+					cpDefinition.getCommerceCatalog();
+
+				return String.valueOf(
+					_commerceInventoryEngine.getStockQuantity(
+						cpInstance.getCompanyId(), commerceCatalog.getGroupId(),
+						commerceChannel.getGroupId(), cpInstance.getSku()));
+			}
+			else if (field.equals("gtin")) {
+				return cpInstance.getGtin();
+			}
+			else if (field.equals("manufacturerPartNumber")) {
+				return cpInstance.getManufacturerPartNumber();
+			}
+			else if (field.equals("sku")) {
+				return cpInstance.getSku();
+			}
+		}
+		catch (CPDefinitionIgnoreSKUCombinationsException
+					cpDefinitionIgnoreSKUCombinationsException) {
+
+			if (_log.isDebugEnabled()) {
+				_log.debug(cpDefinitionIgnoreSKUCombinationsException);
+			}
+		}
+
+		return StringPool.BLANK;
+	}
+
+	private String _getFieldLabel(
+		FragmentEntryLink fragmentEntryLink, String field) {
+
+		try {
+			JSONObject jsonObject1 = _jsonFactory.createJSONObject(
+				fragmentEntryLink.getConfiguration());
+
+			JSONArray jsonArray1 = jsonObject1.getJSONArray("fieldSets");
+
+			JSONArray jsonArray2 = jsonArray1.getJSONObject(
+				0
+			).getJSONArray(
+				"fields"
+			);
+
+			JSONObject jsonObject2 = jsonArray2.getJSONObject(
+				0
+			).getJSONObject(
+				"typeOptions"
+			);
+
+			JSONArray jsonArray3 = jsonObject2.getJSONArray("validValues");
+
+			for (Object validValueObject : jsonArray3) {
+				JSONObject jsonObject3 = (JSONObject)validValueObject;
+
+				String value = jsonObject3.getString("value");
+
+				if (value.equals(field)) {
+					return jsonObject3.getString("label");
+				}
```

Comment:
Is there any other way to do this?

Reply:
Unfortunately I didn't found another way to get the "label" of the selected item from the configuration. I have to parse the json to find the selected option by the "selected value"

Reply:
Done